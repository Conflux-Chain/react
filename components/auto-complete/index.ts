import AutoComplete, { useAutoCompleteHandle } from './auto-complete'
export { useAutoCompleteHandle }
export default AutoComplete
