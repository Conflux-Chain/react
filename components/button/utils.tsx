import React, { ReactNode } from 'react'
import { NormalSizes } from '../utils/prop-types'
import ButtonIcon from './button-icon'
import { ButtonProps } from './button'
import { ButtonGroupConfig } from '../button-group/button-group-context'

export const getButtonChildrenWithIcon = (
  loading: boolean,
  auto: boolean,
  size: NormalSizes,
  children: ReactNode,
  icons: {
    icon?: React.ReactNode
    iconRight?: React.ReactNode
  },
  isTag: boolean,
) => {
  const { icon, iconRight } = icons
  const hasIcon = icon || iconRight
  const isRight = Boolean(iconRight)
  const paddingForAutoMode =
    auto || size === 'mini' || isTag
      ? `calc(var(--zeit-ui-button-height) / 2 + var(--zeit-ui-button-padding) * .5)`
      : 0
  if (!hasIcon) return <div className={`text ${loading ? 'hidden' : ''}`}>{children}</div>
  return (
    <>
      <ButtonIcon isRight={isRight}>{hasIcon}</ButtonIcon>
      <div className={`text ${isRight ? 'right' : 'left'} ${loading ? 'hidden' : ''}`}>
        {children}
        <style jsx>{`
          .left {
            padding-left: ${paddingForAutoMode};
          }
          .right {
            padding-right: ${paddingForAutoMode};
          }
        `}</style>
      </div>
    </>
  )
}

export const filterPropsWithGroup = (
  props: React.PropsWithChildren<ButtonProps>,
  config: ButtonGroupConfig,
): ButtonProps => {
  if (!config.isButtonGroup) return props
  return {
    ...props,
    auto: true,
    shadow: false,
    ghost: config.ghost || props.ghost,
    size: config.size || props.size,
    variant: config.variant || props.variant,
    color: config.color || props.color,
    disabled: config.disabled || props.disabled,
  }
}
