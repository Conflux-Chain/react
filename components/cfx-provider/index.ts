import CfxProvider from './cfx-provider'

export default CfxProvider
