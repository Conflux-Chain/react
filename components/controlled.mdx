If you think the data flow in react carefully.

uncontrolled component:
caller do not know the internal state of the component

controlled component:
the state of called component is fully controlled by `props` passed in 

The above two are identified by react offical team. However, there is also something 
in between, which can not fit in the patterns mentioned above

third patten: the called need to know the callee state sometims but not always
think form as an example, the caller only need to know the state of callee when 'submit'
happen rather than keep track of the component all the time

forth pattern: the caller need to set the callee state forcelly on a one time, not inter
