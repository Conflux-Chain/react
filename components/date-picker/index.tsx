import { Moment } from 'moment'
import momentGenerateConfig from '@jnoodle/rc-picker/lib/generate/moment'
import generatePicker, {
  PickerProps,
  PickerDateProps,
  RangePickerProps as BaseRangePickerProps,
} from './generate-picker'

export type DatePickerProps = PickerProps<Moment>
export type MonthPickerProps = Omit<PickerDateProps<Moment>, 'picker'>
export type WeekPickerProps = Omit<PickerDateProps<Moment>, 'picker'>
export type RangePickerProps = BaseRangePickerProps<Moment>

export const DatePicker = generatePicker<Moment>(momentGenerateConfig)

export default DatePicker
