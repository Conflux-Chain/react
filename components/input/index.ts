import Input, { useInputHandle } from './input'

export { useInputHandle }
export default Input
