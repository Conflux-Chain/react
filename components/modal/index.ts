import Modal from './modal'
import ModalTitle from './modal-title'
import ModalSubtitle from './modal-subtitle'
import ModalContent from './modal-content'
import ModalAction from './modal-action'
import useModalHandle from './use-imperative-modal'

Modal.Title = ModalTitle
Modal.Subtitle = ModalSubtitle
Modal.Content = ModalContent
Modal.Action = ModalAction
Modal.useModalHandle = useModalHandle

export default Modal
