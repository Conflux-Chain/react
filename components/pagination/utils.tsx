export const getPageCount = (total: number, pageSize: number) => {
  return Math.floor((total - 1) / pageSize) + 1
}
