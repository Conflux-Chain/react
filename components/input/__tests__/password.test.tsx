import React from 'react'
import { mount } from 'enzyme'
import { Input } from 'components'
import { nativeEvent } from 'tests/utils'

describe('InputPassword', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <div>
        <Input.Password />
        <Input.Password solid />
      </div>,
    )
    let el = wrapper.find('input').at(0).getDOMNode() as HTMLInputElement
    expect(el.type).toEqual('password')
    el = wrapper.find('input').at(1).getDOMNode() as HTMLInputElement
    expect(el.type).toEqual('password')
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should forward ref by default', () => {
    const ref = React.createRef<HTMLInputElement>()
    const wrapper = mount(<Input.Password ref={ref} />)
    expect(ref.current).not.toBeNull()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should toggle input type', () => {
    const wrapper = mount(<Input.Password />)
    wrapper.find('.input-icon').simulate('click', nativeEvent)
    const el = wrapper.find('input').getDOMNode() as HTMLInputElement
    expect(el.type).toEqual('text')
  })

  it('should hide toggle icon', () => {
    const wrapper = mount(<Input.Password hideToggle />)
    expect(wrapper.find('.input-icon').length).toBe(0)
  })
})
