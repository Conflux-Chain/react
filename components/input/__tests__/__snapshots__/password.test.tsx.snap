// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPassword should render correctly 1`] = `
"<div class=\\"with-label\\"><div class=\\"input-container \\"><div class=\\"input-wrapper lined    \\"><input type=\\"password\\" class=\\"right-icon\\" placeholder=\\"\\" autocomplete=\\"off\\" value=\\"\\"><span class=\\"input-icon\\"><svg viewBox=\\"0 0 24 24\\" width=\\"16\\" height=\\"16\\" stroke=\\"currentColor\\" stroke-width=\\"1.5\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" fill=\\"none\\" shape-rendering=\\"geometricPrecision\\" style=\\"color: currentColor;\\"><path d=\\"M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z\\"></path><circle cx=\\"12\\" cy=\\"12\\" r=\\"3\\"></circle></svg><style>
        .input-icon {
          box-sizing: content-box;
          display: flex;
          width: calc(1.687 * 16px * .42);
          height: 100%;
          align-items: center;
          vertical-align: middle;
          margin: 0;
          padding: 0 calc(1.687 * 16px * .3);
          line-height: 1;
          position: relative;
          cursor: pointer;
          pointer-events: auto;
        }
      </style></span></div></div><style>
          .with-label {
            display: inline-block;
            width: initial;
            box-sizing: border-box;
            -webkit-box-align: center;
          }

          .input-container {
            display: inline-flex;
            align-items: center;
            width: initial;
            height: calc(1.687 * 16px);
          }

          .input-wrapper {
            display: inline-flex;
            vertical-align: middle;
            align-items: center;
            height: 100%;
            flex: 1;
            user-select: none;
            border-radius: 4px;
            transition: border 0.2s ease 0s, color 0.2s ease 0s;
            background-color: transparent;
            border: 1px solid #ebeced;
          }

          .input-wrapper.left-label {
            border-top-left-radius: 0;
            border-bottom-left-radius: 0;
          }

          .input-wrapper.right-label {
            border-top-right-radius: 0;
            border-bottom-right-radius: 0;
          }

          .input-wrapper.disabled {
            cursor: not-allowed;
          }

          .lined.input-wrapper.disabled {
            border-color: #ebeced;
          }

          .solid.input-wrapper.disabled {
            background-color: #dbdde0;
          }

          .focus.input-wrapper:not(.disabled),
          .hover.input-wrapper:not(.disabled) {
            border-color: #0054fe;
            background-color: inherit;
          }

          input {
            margin: 0 1.1429rem;
            padding: 0;
            box-shadow: none;
            font-size: 1rem;
            line-height: calc(1.875 * 16pt - 2px);
            background-color: transparent;
            font-weight: normal;
            border: none;
            color: #282D30;
            outline: none;
            border-radius: 0;
            width: 100%;
            min-width: 0;
            -webkit-appearance: none;
          }

          input:disabled {
            cursor: not-allowed;
            color: #aaafbb;
          }

          input.left-icon {
            margin-left: 0;
          }

          input.right-icon {
            margin-right: 0;
          }

          .input-wrapper.focus:not(.disabled) input,
          .input-wrapper.hover:not(.disabled) input {
            color: #0054fe;
          }

          ::placeholder,
          ::-moz-placeholder,
          :-ms-input-placeholder,
          ::-webkit-input-placeholder {
            color: #9b9eac;
          }

          input::-moz-placeholder {
            line-height: calc(1.875 * 16pt - 1px);
          }

          input:-webkit-autofill,
          input:-webkit-autofill:active {
            -webkit-box-shadow: 0 0 0 30px transparent inset !important;
            -webkit-text-fill-color: #282D30 !important;
          }

          input:-webkit-autofill:hover,
          input:-webkit-autofill:focus {
            -webkit-box-shadow: 0 0 0 30px inherit inset !important;
            -webkit-text-fill-color: #0054fe !important;
          }

          input:-webkit-autofill,
          input:-webkit-autofill:hover,
          input:-webkit-autofill:focus,
          input:-webkit-autofill:active {
            -webkit-background-clip: text;
          }
        </style></div>"
`;
